### ISSUE TITLE
Make sure all variables are initialized in a SCREAMv1 simulation

### ISSUE THREAD

**Issue by PeterCaldwell** [E3SM-Project/scream#838]
When we start a SCREAMv1 run (in "initial" or "restart" format), we should confirm that variables aren't used before they are initialized. I suspect this will be managed through AD code magic related to the DAG?

**Comment by bartgol** [comment]
Yes, the AD DAG inspection should take care of verifying that every input field comes from a valid source. A valid source can be a `FieldInitializer` class (only for standalone scream), surface coupling, or a scorpio input stream.

**Comment by PeterCaldwell** [comment]
Our DAG allows us to confirm that variables are getting updated before being used *in a typical* timestep. It won't confirm that variables are being assigned values in the first step of a simulation. This probably isn't a problem for restart runs, which should just read in all the info they need to continue like a typical step. It is a problem for the first step of a run, where initial conditions are being pulled in from all sorts of places. I'm not sure how to test for this. Is there a compile-time flag that assigns all new arrays to NaN for running in debug mode?

**Comment by bartgol** [comment]
> Our DAG allows us to confirm that variables are getting updated before being used _in a typical_ timestep. It won't confirm that variables are being assigned values in the first step of a simulation. This probably isn't a problem for restart runs, which should just read in all the info they need to continue like a typical step. It is a problem for the first step of a run, where initial conditions are being pulled in from all sorts of places. I'm not sure how to test for this. Is there a compile-time flag that assigns all new arrays to NaN for running in debug mode?

We can make the field repo assign NaN to all field upon allocation, at least in debug mode. That should be an easy (and likely healthy) thing to do.

**Comment by bartgol** [comment]
I think this is a combo of #836 and #837 , since if required data is read in, then input variables _are_ initialized. I'm gonna close this.

**Comment by bartgol** [comment]
I think the AD is already achieving the result, albeit maybe from a different angle. The AD grabs all the input to the atm proc sequence (i.e., all inputs of the 1st process, plus all inputs from the 2nd that are not computed by the 1st, and so on). Then it tries to initialize all these inputs (either read from nc IC file, or set constant values as per user request in the yaml file). If any of these inputs are not inited manually (from yaml file) and not found in the nc file, then someone will throw an error (I think it might be scorpio that actually throws).

So, in short, scream won't complete the initialization step if there are inputs that are not correctly initialized. Which I think is what we need/want in the end?

**Comment by PeterCaldwell** [comment]
Ok, yeah that sounds perfect. Ok, we can close this... wait - you already did ;-).
